osf/migrations/0155_add_registration_files_count.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-01-18 14:43
+from __future__ import unicode_literals
+import logging
+
+from django.db import migrations, models
+from django_bulk_update.helper import bulk_update
+
+logger = logging.getLogger(__file__)
+
+def add_registration_files_count(state, *args, **kwargs):
+    AbstractNode = state.get_model('osf.abstractnode')
+    BaseFileNode = state.get_model('osf.basefilenode')
+    ContentType = state.get_model('contenttypes', 'ContentType')
+    registrations = AbstractNode.objects.filter(type='osf.registration', is_deleted=False)
+    content_type = ContentType.objects.get(app_label='osf', model='abstractnode')
+
+    registrations_to_update = []
+    for registration in registrations:
+        job = registration.archive_jobs.first() if registration.archive_jobs.count() else None
+        archiving = job and not job.done and (job.status != 'SUCCESS')
+        if archiving:
+            # Skip stuck, failed, or archiving registrations.
+            continue
+        registration_files = BaseFileNode.objects.filter(
```

Comment:
This all looks equivalent to:
```py
registration.files.filter(deleted_on=None).count()
```
Could we do that instead or is there something I'm missing?

Reply:
Django only uses [historical models](https://docs.djangoproject.com/en/2.1/topics/migrations/#historical-models) during migrations, hence `.files` is not available.

Reply:
`files` is a django field, though, not a custom property. Aren't those available on historical models?
https://github.com/CenterForOpenScience/osf.io/blob/44ce29fa8e969a3fdb1e982c9983422ca7113d25/osf/models/node.py#L327

Reply:
I suspect that it's because `GenericRelation` just like `GenericForeignKey` fields are not normal object ie. not accessible through the django DB API. They are just preferred over normal foreign key to describe `polymorphic` relationships.

Reply:
Huh, ok, you're right. Still seems weird to me, but I guess that's how it works 👍